### ISSUE TITLE
Annotation is present on method if {there is another annotation|method signature pattern}

### ISSUE THREAD

**Issue by dgroup** [dgroup/arch4u-pmd#28]
### Overview
Each application should provide metrics for future application monitoring, especially if its expose API endpoints.

### The rule
If we expose a particular REST endpoint, this endpoint should be monitored as well, therefore, each method with `@Get`, `@Post`, etc. annotations should have annotation `@Timed` with metric reference.

`@Timed` is just an example, in the future we'll configure several REST frameworks with different standards for monitoring (e.g micrometer, etc.).

### Ruleset(s)
- [ ] arch4u-micrometer-ruleset.xml (with monitoring related test cases e.g. micrometer `@Timed` annotation)
- [ ] arch4u-template-ruleset.xml (with non-monitoring related test cases e.g. any other annotation like `@Override`)

### Test case 1 [positive/right]: Expected annotation is present on the method that has another annotation
```java

@RestController
public class MyTypicalRestController {
   
   @Timed                                          
   @GetMapping("/get")                 // ok, @Timed is present for all methods with annotation @GET
   public .. get(BigInteger id) {
      ..
   }

   @GetMapping("/getXXX")              // violation, no annotation @Timed found
   public .. getXXX(BigInteger id) {
      ..
   }
}
```

### Test case 2 [positive/right]: Expected annotation is present on the method that has some signature like `public Order getXXXX`
```java

@RestController
public class MyTypicalRestController {
   
   @Timed                                          
   @GetMapping("/get")                 // ok, @Timed is present for all methods with annotation @GET
   public .. get(BigInteger id) {
      ..
   }

   @GetMapping("/getXXX")              // violation, no annotation @Timed found for method `public Order getXXXX`
   public .. getXXX(BigInteger id) {
      ..
   }
}
```

**Comment by dykov** [comment]
@dgroup please provide some examples with right/wrong cases

**Comment by dykov** [comment]
@dgroup , please describe the second example. why the method with this signature should have `@Timed` annotation?
in this case, it will be better to split the rule into 2 separate rules - it will be easier to specify property values for each rule separately

in my opinion, methods should have such annotation if the class has `@RestController` annotation

**Comment by dgroup** [comment]
@dykov not all methods from `@RestController` will have @Get or @Post annotations, thus its better to inspect each method separately.
